sdk/core/azure-core-amqp/src/impl/rust_amqp/amqp/connection.cpp

```diff
@@ -209,62 +209,70 @@ namespace Azure { namespace Core { namespace Amqp { namespace _detail {
     m_connectionOptions.reset(amqpconnectionoptionsbuilder_build(builder.get()));
   }
 
-  void ConnectionImpl::Open()
+  void ConnectionImpl::Open(Azure::Core::Context const& context)
   {
+    Azure::Core::Amqp::Common::_detail::CallContext callContext{
+        Azure::Core::Amqp::Common::_detail::GlobalStateHolder::GlobalStateInstance()
+            ->GetRuntimeContext(),
+        context};
     if (m_options.EnableTrace)
     {
       Log::Stream(Logger::Level::Verbose)
           << "ConnectionImpl::Open: " << this << " ID: " << m_containerId;
     }
     if (amqpconnection_open(
-            Azure::Core::Amqp::Common::_detail::GlobalStateHolder::GlobalStateInstance()
-                ->GetRuntimeContext(),
+            callContext.GetCallContext(),
             m_connection.get(),
             m_hostUrl.GetAbsoluteUrl().c_str(),
             m_containerId.c_str(),
             m_connectionOptions.get()))
     {
-      auto error = runtime_context_get_error(
-          Azure::Core::Amqp::Common::_detail::GlobalStateHolder::GlobalStateInstance()
-              ->GetRuntimeContext());
-      auto err = rust_error_get_message(error);
-      std::string errorString{err};
-      rust_string_delete(err);
-      rust_error_delete(error);
-      throw std::runtime_error("Could not open connection:" + errorString);
+      throw std::runtime_error("Could not open connection:" + callContext.GetError());
```

Comment:
We probably need an extra space there
```suggestion
      throw std::runtime_error("Could not open connection: " + callContext.GetError());
```